File: app/src/test/java/com/example/mike/mp3player/client/views/viewholders/MediaItemViewHolderTestBase.java
package com.example.mike.mp3player.client.views.viewholders;

import android.view.View;

import com.example.mike.mp3player.client.AlbumArtPainter;

import org.mockito.Mock;

public abstract class MediaItemViewHolderTestBase<V extends MediaItemViewHolder> {

    @Mock
    View view;
    @Mock
    AlbumArtPainter albumArtPainter;

    V mediaItemViewHolder;


}

File: client/src/test/java/com/github/goldy1992/mp3player/client/views/viewholders/MediaItemViewHolderTestBase.kt
package com.github.goldy1992.mp3player.client.views.viewholders

import android.view.View
import com.github.goldy1992.mp3player.client.AlbumArtPainter
import com.nhaarman.mockitokotlin2.mock

abstract class MediaItemViewHolderTestBase<V : MediaItemViewHolder> {

    lateinit var view: View
    lateinit var albumArtPainter: AlbumArtPainter
    lateinit var mediaItemViewHolder: V

    open fun setup() {
        view = mock<View>()
        albumArtPainter = mock<AlbumArtPainter>()
    }
}